src/passes/OptimizeInstructions.cpp

```diff
@@ -360,6 +361,13 @@ struct OptimizeInstructions
             curr->cast<Binary>()->right = y;
             return curr;
           }
+          // i32(x) <<>> (y & 32)   ==>   x
+          // i64(x) <<>> (y & 64)   ==>   x
```

Comment:
These comments might be clearer as `(y & ~31)` to indicate the lower bits are all zero. But even that is kind of confusing because there is no `&` operation actually done. Maybe
```
i32(x) <<>> y   ==>   x   where y == y & ~31
```
?

Reply:
how about `i32(x) <<>>( y & (C & ~31))` ?

Reply:
@MaxGraey Yes, that looks like it would be clearer!

Reply:
https://github.com/WebAssembly/binaryen/pull/3307/commits/37a5dea62e3de271f1ed7ec8f92668ac9bb17220

Reply:
Looks good, thanks! Is that part of a PR? The github UI doesn't make that clear sadly.

Reply:
> Looks good, thanks! Is that part of a PR?

Yes. It's part of #3307  pr